### DISCUSSION TITLE
What is the difference between UserLocation and LocationPuck ?

### DISCUSSION THREAD

**Discussion by TheUncharted** [rnmapbox/maps#3325] [Q&A] [answered]
I was wondering what are the difference between both components ? 
They are both rendering a circle, UserLocation has an onUpdate that is really useful but LocationPuck  does not but seems to have more customization capabilities ?
What are the use cases for them ? Can or should we use both ?

**Comment by mfazekas** [comment]
LocationPuck is native while UserLocation is react-native. So UserLocation causes a pre-render on each location update. So if LocationPuck works for you that's preferred for performance reasons, but UserLocation should be ok as well. On longer term we might (or might not) deprecate UserLocation in favour of LocationPuck.

**Reply by TheUncharted** [reply]
is it possible to have the onUpdate on the LocationPuck ? or is there another way to do it ?

**Reply by mfazekas** [reply, marked as answer]
You can get the current location via locationManager 

https://github.com/rnmapbox/maps/blob/main/src/modules/location/locationManager.ts

**Reply by nixolas1** [reply]
Awesome! Was looking for a way to do locationUpdates with LocationPucks and v11 mapbox.
Could be nice to have documented elsewhere ;)
